docs/how-to-write-a-contract/built-ins.md

```diff
@@ -354,6 +354,46 @@ const unsignedTx: bsv.Transaction = new bsv.Transaction()
   .change(options.changeAddress);
 ```
 
+### `insertCodeSeparator`
+
+Method `insertCodeSeparator(): void` will insert an [`OP_CODESEPARATOR`](https://wiki.bitcoinsv.io/index.php/OP_CODESEPARATOR), where it is invoked.
+
+```ts
+export class OCS extends SmartContract {
+
+    @prop()
+    readonly pubKeyHash: PubKeyHash;
+
+    constructor(pubKeyHash: PubKeyHash) {
+        super(pubKeyHash);
+        this.pubKeyHash = pubKeyHash;
+    }
+
+    @method()
+    public unlock(sig0: Sig, sig1: Sig pubkey: PubKey) {
+        assert(hash160(pubkey) == this.pubKeyHash, 'pubKeyHash check failed');
+        this.insertCodeSeparator()
+        assert(this.checkSig(sig0, pubkey));
+        this.insertCodeSeparator()
+        assert(this.checkSig(sig1, pubkey));
+    }
+
+}
+```
+
+In the above example, the `unlock` method calls `insertCodeSeparator`. This implies that each invocation of `checkSig` will use the code below the most recent invocation of `insertCodeSeparator` within the signature verification process. This includes the invocation of `insertCodeSeparator` itself.
+
+Therefore, this functionality requires us to modify the script when creating a signature. This is so that the signature signs only the necessary subscript, instead of the entire locking script. We can achieve this script alteration using the [`subScript`](../reference/classes/bsv.Script-1.md#subscript) function, which returns the subscript, cut at the n-th occurence of `OP_CODESEPARATOR`:
+
+```ts
+const tx = newTx();
+let demo = new OCS(PubKeyHash(toHex(pkh)))
+demo.to = { tx, inputIndex }
+
```

Comment:
Do this after #404.
Move all this to a separate doc named OP_CodeSeparator under https://docs.scrypt.io/category/advanced. In it, explain

1. how checkSig works https://en.bitcoin.it/wiki/OP_CHECKSIG#How_it_works and thus how OCS works. Use diagram here https://xiaohuiliu.medium.com/op-codeseparator-97dc039a8ec8
2. show the example contract: OCS. Rename it to OpodeSeparator, to be more clear.